freqtrade/persistence/trade_model.py

```diff
@@ -1418,7 +1450,7 @@ def get_open_order_trades() -> List['Trade']:
         Returns all open trades
         NOTE: Not supported in Backtesting.
         """
-        return cast(List[Trade], Trade.get_trades(Trade.open_order_id.isnot(None)).all())
+        return cast(List[Trade], Trade.get_trades([Trade.open_orders_count != 0]).all())
```

Comment:
This is wrong, and causes at least one test to fail (namely `test_cancel_all_open_orders`), and can potentially lead to quite some inefficiencies, maybe even bugs or oddities when it comes to orders going "missing" (or orders being processed multiple times).

`Trade.open_orders_count` is a property - so while this looks good at first - the resulting query is the following (notice the empty where clause).


```
'SELECT trades.id, trades.exchange, trades.pair, trades.base_currency, trades.stake_currency, trades.is_open, trades.fee_open, trades.fee_open_cost, trades.fee_open_currency, trades.fee_close, trades.fee_close_cost, trades.fee_close_currency, trades.open_rate, trades.open_rate_requested, trades.open_trade_value, trades.close_rate, trades.close_rate_requested, trades.realized_profit, trades.close_profit, trades.close_profit_abs, trades.stake_amount, trades.max_stake_amount, trades.amount, trades.amount_requested, trades.open_date, trades.close_date, trades.stop_loss, trades.stop_loss_pct, trades.initial_stop_loss, trades.initial_stop_loss_pct, trades.stoploss_order_id, trades.stoploss_last_update, trades.max_rate, trades.min_rate, trades.exit_reason, trades.exit_order_status, trades.strategy, trades.enter_tag, trades.timeframe, trades.trading_mode, trades.amount_precision, trades.price_precision, trades.precision_mode, trades.contract_size, trades.leverage, trades.is_short, trades.liquidation_price, trades.interest_rate, trades.funding_fees \nFROM trades \nWHERE true'
```

Reply:
You are right this is not good and also for another reason "Trade.open_orders_count != 0 " is not similar to "Trade.open_order_id.isnot(None)" as It will also return stoploss orders.

Reply:
Do you think Trade.is_open.is_(True) filter could be good? or we need to checks withing orders array of Trade